openedx/core/djangoapps/programs/utils.py

```diff
@@ -462,27 +478,17 @@ def _collect_one_click_purchase_eligibility_data(self):
         bundle_variant = 'full'
         if is_learner_eligible_for_one_click_purchase:
             for course in self.data['courses']:
-                add_course_sku = False
-                published_course_runs = filter(lambda run: run['status'] == 'published', course['course_runs'])
+                add_course_sku = True
+                course_runs = course.get('course_runs', [])
+                published_course_runs = filter(lambda run: run['status'] == 'published', course_runs)
+
```

Comment:
I think you meant to add course_runs and published_course_runs together for use as course_runs in the loop below